### ISSUE TITLE
Knockout 3.5.0 <select> binding doesn't notify on initial binding anymore when the options are generated via a foreach binding

### ISSUE THREAD

**Issue by Zallist** [knockout/knockout#2452]
Simplified example below. The actual scenario involves generating ``<optgroup>`` and ``<option>`` elements based on a ``foreach`` binding.

In essence, in Knockout 3.4.2 you were previously able to put a ``foreach`` binding inside a ``<select>`` element to generate the ``<option>`` elements and Knockout worked fine with it. In Knockout 3.5.0 the ``<option>`` elements are still generated fine, but the ``<select>`` does not show a selected item after initial load anymore.

HTML:
```html
<div>
  <select data-bind="value: selectedValue, valueAllowUnset: true">
    <option>-- Please Select--</option>
    <!-- ko template: { foreach: [5] } -->
    <optgroup label="group 1">
      <option value="group 1 item 1">group 1 item 1</option>
      <option value="group 1 item 2">group 1 item 2</option>
    </optgroup>
    <optgroup label="group 2">
      <option value="group 2 item 1">group 2 item 1</option>
      <option value="group 2 item 2">group 2 item 2</option>
    </optgroup>
    <!-- /ko -->
  </select>
</div>
```

JavaScript:
```javascript
(function () {
  var vm = {
  	selectedValue: ko.observable('group 1 item 2')
  };

  ko.applyBindings(vm);
})();
```

JSFiddle with Knockout 3.5.0: https://jsfiddle.net/2cbaoe4r/5/
JSFiddle with Knockout 3.4.2: https://jsfiddle.net/2cbaoe4r/6/

**Comment by Zallist** [comment]
Potentially broke by the commit https://github.com/knockout/knockout/commit/bd6e50879b0778d64977570ac26fdf094769f25e#diff-3745d5a97955abda7d49a8be150125d4

It used to always, on a setTimeout, set the value. So even though the inner elements render after the select is bound, it then just resets the value a second time.
This got changed to only happen in ie6, though.

**Comment by fastfasterfastest** [comment]
Yes, this is a problem.

>It used to always, on a setTimeout, set the value.

Actually, it used to always set the value twice - "once directly" and then a 2nd time in `setTimeout`.  Now, in 3.5.0, it only sets the value once.
Which is fine and ok... unless you are using a `foreach` binding to generate `option` elements.
The `foreach` processing causes the select element to "loose" its value, both in 3.4.2 and in 3.5.0, but in 3.4.2 the 2nd call to `ko.selectExtensions.writeValue` re-set it and saved us - that no longer happens in 3.5.0 (unless you are on IE6...)

**Comment by mbest** [comment]
A workaround is to add `if: true` to the binding. https://jsfiddle.net/mbest/1qo9tsg3/

**Comment by chrisknoll** [comment]
Wonder if this is something where 'descendantsComplete' would be useful: it's a case where we need to signal the parent `value` binding that the descendants (the foreach and template binding) have completed.

**Comment by QDCSLG** [comment]
Would it help to add the selected attribute binding?

<div>
  <select data-bind="value: selectedValue, valueAllowUnset: true">
    <option>-- Please Select--</option>
    
    <optgroup label="group 1">
      <option value="group 1 item 1"
*data-bind="attr:{'selected':$data==$parent.selectedValue()}"*>group 1
item 1</option>
      <option value="group 1 item 2">group 1 item 2</option>
    </optgroup>
    <optgroup label="group 2">
      <option value="group 2 item 1">group 2 item 1</option>
      <option value="group 2 item 2">group 2 item 2</option>
    </optgroup>
    
  </select>
</div>

**Comment by mbest** [comment]
@QDCSLG No, that won't help.

**Comment by mbest** [comment]
I've uploaded a fix in #2460. I'll wait to see if anyone has any concerns or feedback before merging.
